### ISSUE TITLE
While loops

### ISSUE THREAD

**Issue by KittyGiraudel** [mknadler/randomize.scss#7]
Any reason you prefer using while loops rather than for loops? I find them harder to read and maintain, but perhaps I missed something.

**Comment by mknadler** [comment]
I suppose that, in theory, I prefer `@while` for more abstract cases, such as in `@function shuffle`, and especially when the loop isn't actually outputting CSS particular to each pass through the loop, but is compiling a single value to be returned. By comparison, I generally think of `@for` as being for when CSS is being directly outputted within each pass, and/or the output's change is dramatically changed each time.

In short: `@while` for "do the same thing six times" (`random-hex`), `@for` for "do six things, the nature of which differs depending on `$i`".

In practice, reviewing my code, I apply these principle somewhat inconsistently, and will probably adjust the code so that I am using the different loop styles in a consistent manner.

**Comment by KittyGiraudel** [comment]
Alright. As long as your consistent, I suppose that's okay. Note that (Sass) developers usually find while loops harder to use than for loops. Also, I do think while loops have nothing to do in Sass, especially since there is no way to break the loop from the inside.

**Comment by mknadler** [comment]
Noted. I will likely rewrite in the future with this in mind. 

The _reason_ for concern about `@while loops` not breaking from inside -- or at least, _one_ of the reasons --  just clicked with me this morning: if you don't account for a particular kind of input with `@warn`s or `@error`s, and it goes into a `@for` loop, it'll be fine -- that part of the stylesheet may not compile correctly. With a `@while` loop, it'd all break.
